parser-typechecker/unison/Options.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Options where
+
+import Control.Applicative ((<**>), (<|>), some)
+import Data.Semigroup ((<>))
+import Options.Applicative
+  ( Parser,
+    ParserInfo,
+    command,
+    footerDoc,
+    help,
+    helper,
+    hsubparser,
+    info,
+    long,
+    metavar,
+    optional,
+    progDesc,
+    strArgument,
+    strOption,
+    switch,
+  )
+import Text.PrettyPrint.ANSI.Leijen ((<+>), (</>))
+
+data Options
+  = Options
+      { codepath :: Maybe FilePath,
+        cmd :: Command
+      }
+  deriving (Show)
+
+data Command
+  = Launch
+  | Version
+  | Init
+  | Run (Maybe FilePath) Bool String
+  | Transcript Bool Bool [FilePath]
```

Comment:
Could use a comment to understand these args inline

Reply:
newtypes!